File: crates/nil-payload/src/cheat/infrastructure.rs
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

use nil_core::continent::Coord;
use nil_core::infrastructure::prelude::{BuildingId, BuildingLevel};
use nil_core::ruler::Ruler;
use nil_core::world::WorldId;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct CheatGetAcademyRecruitQueueRequest {
  pub world: WorldId,
  pub coord: Coord,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct CheatGetInfrastructureRequest {
  pub world: WorldId,
  pub coord: Coord,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct CheatGetPrefectureBuildQueueRequest {
  pub world: WorldId,
  pub coord: Coord,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct CheatGetStableRecruitQueueRequest {
  pub world: WorldId,
  pub coord: Coord,
}

#[derive(Clone, Debug, Default, Deserialize, Serialize)]
#[serde(default, rename_all = "camelCase")]
pub struct CheatGetStorageCapacityRequest {
  pub world: WorldId,
  pub ruler: Option<Ruler>,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct CheatSetBuildingLevelRequest {
  pub world: WorldId,
  pub coord: Coord,
  pub id: BuildingId,
  pub level: BuildingLevel,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct CheatSetMaxInfrastructureRequest {
  pub world: WorldId,
  pub coord: Coord,
}

File: app/src/lib/request/cheat/infrastructure.ts
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

export interface CheatGetAcademyRecruitQueueRequest {
  readonly world: WorldId;
  readonly coord: Coord;
}

export interface CheatGetAcademyRecruitQueuesRequest {
  readonly world: WorldId;
  readonly coords: readonly Coord[];
  readonly filterEmpty: boolean;
}

export interface CheatGetAllAcademyRecruitQueuesRequest {
  readonly world: WorldId;
  readonly filterEmpty: boolean;
}

export interface CheatGetAllPrefectureBuildQueuesRequest {
  readonly world: WorldId;
  readonly filterEmpty: boolean;
}

export interface CheatGetAllStableRecruitQueuesRequest {
  readonly world: WorldId;
  readonly filterEmpty: boolean;
}

export interface CheatGetInfrastructureRequest {
  readonly world: WorldId;
  readonly coord: Coord;
}

export interface CheatGetPrefectureBuildQueueRequest {
  readonly world: WorldId;
  readonly coord: Coord;
}

export interface CheatGetPrefectureBuildQueuesRequest {
  readonly world: WorldId;
  readonly coords: readonly Coord[];
  readonly filterEmpty: boolean;
}

export interface CheatGetStableRecruitQueueRequest {
  readonly world: WorldId;
  readonly coord: Coord;
}

export interface CheatGetStableRecruitQueuesRequest {
  readonly world: WorldId;
  readonly coords: readonly Coord[];
  readonly filterEmpty: boolean;
}

export interface CheatGetStorageCapacityRequest {
  readonly world: WorldId;
  readonly ruler: Option<Ruler>;
}

export interface CheatSetBuildingLevelRequest {
  readonly world: WorldId;
  readonly coord: Coord;
  readonly id: BuildingId;
  readonly level: BuildingLevel;
}

export interface CheatSetMaxInfrastructureRequest {
  readonly world: WorldId;
  readonly coord: Coord;
}
